docs/howto/danceable-lyrics-input-tool/README.md

```diff
@@ -0,0 +1,485 @@
+When you have completed the above steps, your `__init__` should look something like the following:
+
+```python
+def __init__(self, provider: AMPProviderV2) -> None:
+    """Construct a plugin."""
+    self.provider = provider
+    self.name = "DanceableLyrics"
+
+    # To build a more flexible plugin, read input path(s) from the user
+    self.DATASETS_BASE = "c:/users/bryan.ashby/DanceableLyricsData/"
+
+    self.provider.push_outgoing_metadata(
+        "Output",
+        create_schema(
+            {
+                "artist_name": {"type": FieldType.v_wstring,},
+                "track_name": {"type": FieldType.v_wstring,},
+                "danceability": {"type": FieldType.float,},
+                "energy": {"type": FieldType.float,},
+                "track_id": {"type": FieldType.v_wstring,},
+            }
+        ),
+    )
+
+    self.LYRICS_TERMS = [
+        "star wars",
+        "star trek",
+        "luke skywalker",
+        "captain kirk",
+        "spock",
+        "yoda",
+    ]
+    self.DANCEABILITY_RANGE = [0.45, 0.99]
+    self.ENERGY_RANGE = [0.45, 0.75]
+    self.TEMPO_RANGE = [110.0, 140.0]
+    self.MIN_VIEWS = 1000
+
+    self.provider.io.info(f"{self.name} initialized.")
+```
+
+> :information_source: To make your Input plugin more dynamic, many of the variables above can be made to be read in from the user using the [Alteryx UI-SDK](https://github.com/alteryx/dev-harness)!
+
+## 4. Data Processing
+Now it's time to process some data! Since we do not currently attempt incoming data, but rather, _provide_ data as output, we'll be doing the bulk of our work in `on_complete`.
+
+Our first data input will be from `genius_song_lyrics.csv` in which we will discover songs with lyrics containing our search terms:
+
+```python
+def on_complete(self) -> None:
+    self.provider.io.info(f"{self.name} gathering sample lyrics...")
+
+    sample = (
+        pl.scan_csv(self.DATASETS_BASE + "genius_song_lyrics.csv")
+        .select(
+            pl.col("title").str.to_lowercase().alias("track_name"),
+            pl.col("artist").str.to_lowercase().alias("artist_name"),
+            "lyrics",
+            "language",
+            "views",
+        )
+        .filter(
+            pl.col("lyrics").str.contains("(?i)" + "|(?i)".join(self.LYRICS_TERMS))
+        )
+        .filter(pl.col("language") == "en")
+        .filter(pl.col("views") > self.MIN_VIEWS)
+        .collect()
+    )
+```
+
+Above: We are telling Polars to create a `DataFrame` containing select columns from rows matching our criteria. We are also normalizing some column names by lowercasing them and aliasing them in our output. In our `filter` clause, we are building a case-insensitive [Regular Expression](https://en.wikipedia.org/wiki/Regular_expression) from our list of terms. Finally, we filter out non-English terms and restrict to only entries with more than `self.MIN_VIEWS`. You may wish to alter this criteria a bit, for example, by allowing other languages.
```

Comment:
TW review:

Above, we tell Polars to create a `DataFrame` that contains select columns from rows that match our criteria. We also normalize some column names— we lowercase them and add aliases in our output. In our `filter` clause, we build a case-insensitive [regular expression](https://en.wikipedia.org/wiki/Regular_expression) from our list of terms. Finally, we filter out non-English terms and restrict results to only entries with more than `self.MIN_VIEWS`. You might want to alter these criteria a bit. For example, you might want to allow other languages.